cli/src/positions/portfolio.tsx

```diff
@@ -235,27 +249,51 @@ export const PortfolioControl = () => {
     addMessage('Terminating scan...')
   }, [addMessage])
 
+  const sync = Boolean(useOption('sync'))
+
+  const duration = useOption('duration')
+
+  useEffect(() => {
+    if (sync && filteredPairs) {
+      onDurationSubmit(duration!)
+    }
+  }, [sync, duration, onDurationSubmit, filteredPairs])
+
+  if (cli && sync) {
```

Comment:
Probably need a loop somewhere to wait until sync is done (e.g. keep printing progress), to prevent node from exiting